Multiplatform/Utility/Intents/SpotlightHelper.swift

```diff
@@ -28,48 +31,63 @@ struct SpotlightHelper {
         
         Task.detached {
             do {
-                let count = try await JellyfinClient.shared.getTracks(limit: 1, startIndex: 0, sortOrder: .name, ascending: true, favorite: false, search: nil).1
-                Defaults[.spotlightDisabled] = count > 10_000
-                
-                if Defaults[.spotlightDisabled] {
-                    return
-                }
-                
                 let index = CSSearchableIndex(name: "items", protectionClass: .completeUntilFirstUserAuthentication)
-                let tracks = try await JellyfinClient.shared.getTracks(limit: 0, startIndex: 0, sortOrder: .name, ascending: true, favorite: false, search: nil).0
-                var items = [CSSearchableItem]()
-                
-                for track in tracks {
-                    let attributes = CSSearchableItemAttributeSet(contentType: .audio)
+                var startIndex = 0
+                if !lastDonationCompleted, let lastData = try? await index.fetchLastClientState(), lastData.count == MemoryLayout<Int>.size {
+                    startIndex = lastData.withUnsafeBytes {
+                        $0.load(as: Int.self).littleEndian
+                    }
+                }
+                if lastDonationCompleted {
```

Comment:
Why is the index deleted?

Reply:
If the last Donation is fully completed and we get into here, it means 48 hours has passed and we need to refresh the index, therefore we have to remove all current indexes and start fresh. This is not the most optimal way but there is no api for tracking library changes yet.